docs/build-aea-programmatically.md

```diff
@@ -71,6 +73,40 @@ Next, we add the echo skill which will bounce our messages back to us. We first
     builder.add_skill("./packages/fetchai/skills/echo")
 ```
 
+Also, we can add a component that was instantiated programmatically. :
+``` python
+    # create skill and handler manually
+    from aea.protocols.base import Message
+    from aea.protocols.default.message import DefaultMessage
+    from aea.skills.base import Handler
+
+    class DummyHandler(Handler):
+        """Dummy handler to handle messages."""
+
+        SUPPORTED_PROTOCOL = DefaultMessage.protocol_id
+
+        def setup(self) -> None:
+            """Noop setup."""
+
+        def teardown(self) -> None:
+            """Noop teardown."""
+
+        def handle(self, message: Message) -> None:
+            """Handle incoming message."""
+            print("Hi, you got a ", message)
+
+    context = SkillContext()
+    config = SkillConfig()
```

Comment:
This is missing the public id elements

Reply:
yes, see my comment above.

I think that would be an argument to introduce heavy consistency checks on the fields of configurations object (and probably many others, like `Message`).

I we agree on that, worth considering in the future projects like Pydantic:  https://github.com/samuelcolvin/pydantic. Never used, but seems a good use case, since we have lots of data classes.

A nice standard package is `dataclasses`: https://docs.python.org/3/library/dataclasses.html. However, it's a pity that's only for Python 3.7+, where we also support Python 3.6.

Reply:
`Message` already has heavy consistency checks. SkillConfig is clearly missing them!

Further dependencies are not allowed at this stage. We are trying to reduce dependencies not add even more. This seems perfectly doable by ourselves.

Reply:
@solarw please, once this PR is merged (https://github.com/fetchai/agents-aea/pull/1180), merge develop and address the above by adding an `author` and a `name`.